src/components/fields/Percentage/Settings.tsx

```diff
@@ -0,0 +1,195 @@
+}) => {
+  const toggleCollapse = () => !disabled && setActive(active ? null : colorKey);
+
+  const toggleState = (() => {
+    if (disabled && active) {
+      setActive(null);
+      return false;
+    }
+    return active;
+  })();
+
+  return (
+    <Box sx={{ width: "100%" }}>
+      <ButtonBase
+        onClick={toggleCollapse}
+        component={ButtonBase}
+        focusRipple
+        disabled={disabled}
+        sx={[
+          fieldSx,
+          {
+            justifyContent: "flex-start",
+            "&&": { pl: 0.75, pr: 0.5 },
+            color: color.hex,
+            transition: (theme) =>
+              theme.transitions.create("border-radius", {
+                delay: theme.transitions.duration.standard,
+              }),
+            "&.Mui-disabled": { color: "text.disabled" },
+          },
+          active && {
+            transitionDelay: "0s",
+            transitionDuration: "0s",
+          },
+        ]}
+      >
+        <Box
+          sx={{
+            backgroundColor: color?.hex,
+            width: 15,
+            height: 15,
+            mr: 1.5,
+            boxShadow: (theme) => `0 0 0 1px ${theme.palette.divider} inset`,
+            borderRadius: 0.5,
+            opacity: 0.5,
+          }}
+        />
+        <div style={{ flexGrow: 1 }}>{color.hex}</div>
+        <ChevronDown
+          color="action"
+          sx={{
+            transition: (theme) => theme.transitions.create("transform"),
+            transform: active ? "rotate(180deg)" : "none",
+          }}
+        />
+      </ButtonBase>
+      <Collapse in={toggleState}>{children}</Collapse>
+    </Box>
+  );
+};
+
+const defaultColors: { [key: string]: Color } = {
+  default: toColor("hex", "#FFFFFF"),
+  startColor: toColor("hex", "#ED4747"),
+  midColor: toColor("hex", "#F3C900"),
+  endColor: toColor("hex", "#1FAD5F"),
+};
+
+const colorLabels: { [key: string]: string } = {
+  startColor: "Start Color",
+  midColor: "Middle Color",
+  endColor: "End Color",
+};
+
+export default function Settings({ onChange, config }: ISettingsProps) {
+  const [checkStates, setCheckStates] = useState<{ [key: string]: boolean }>({
+    startColor: Boolean(config.startColor),
+    midColor: Boolean(config.midColor),
+    endColor: Boolean(config.endColor),
+  });
+  const [activePicker, setActivePicker] = useState<string | null>(null);
+  const theme = useTheme();
+
+  const onCheckboxChange = (key: string, checked: boolean) => {
+    if (!checked) {
+      onChange(key)(defaultColors.default);
+    } else {
+      onChange(key)(defaultColors[key]);
+    }
+    setCheckStates({ ...checkStates, [key]: checked });
+  };
+
+  return (
+    <>
+      {Object.keys(checkStates).map((colorKey) => {
+        const color = config[colorKey] || defaultColors[colorKey];
+        return (
+          <Box key={colorKey} sx={{ display: "flex", flexDirection: "column" }}>
+            {colorLabels[colorKey]}
+            <Box
+              sx={{
```

Comment:
Inline stylings reduce readability here, since this is the most complex part better to use a style wrapper